### PR TITLE
Merge cItemMushroomSoupHandler into cItemFoodHandler

### PR DESCRIPTION
**Pull request by Gargaj**
The main purpose of this is to fix eating the bowl when eating rabbit stew; the current generic food handler is perfectly capable of handling returned items after eating, and an additional class for that single behaviour is unnecessary.

### CODE REVIEW

**Comment by SphinxC0re** [comment]
Looks good to me

File: src/Items/ItemFood.h [hunk 1]
```diff
@@ -101,6 +102,29 @@ class cItemFoodHandler :
...
+
+		switch (m_ItemType)
+		{
+			case E_ITEM_MUSHROOM_SOUP:
+			case E_ITEM_RABBIT_STEW:
+			{
+				// Return a bowl to the inventory
+				if (!a_Player->IsGameModeCreative())
+				{
+					a_Player->GetInventory().AddItem(cItem(E_ITEM_BOWL));
```
Comment (SphinxC0re) [inline src/Items/ItemFood.h]: Shouldn't the bowl go into the same slot the soup has been before?

Comment (Gargaj) [inline src/Items/ItemFood.h]: No idea, I just took the code from the existing ```cItemMushroomSoupHandler```

Comment (SphinxC0re) [inline src/Items/ItemFood.h]: It's actually pretty difficult to do this so your solution is okay

**Comment by madmaxoft** [comment]
Looks good to me, please rebase onto current master and let's get this merged.

**Comment by Gargaj** [comment]
Done.

**Comment by madmaxoft** [comment]
We get some random build failures, I restarted the Travis build. Merge if it succeeds.

**Pull request merged**